Sprint_2

Shim239 opened the pull request:
Добавлены тесты на все классы + отчет о покрытии
Исправлен класс Lion (изолирован, исправлена орфографическая ошибка)

Shim239 committed:
all changes

pomegranate282922 commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,39 @@
+    @Test
+    public void getFoodMethodPositiveResult() throws Exception {
+        Cat cat = new Cat(feline);
+        Mockito.when(cat.getFood()).thenReturn(List.of("Животные", "Птицы", "Рыба"));
```
Здесь ошибочка. Мы никогда не подменяем результат того теста , который сейчас тестируем. Ты в итоге протестировал работу библиотеки Мокито - она действительно подменяет как надо. Подменять нужно поведение зависимостей!

Shim239 replied:
Свою ошибку понял, исправил.

pomegranate282922 commented on src/test/java/com/example/FelineTest.java:
```diff
@@ -0,0 +1,56 @@
+    @Test
+    public void eatMeatMethodPositiveResult() throws Exception {
+        Feline feline = new Feline();
+        Mockito.when(animal.getFood("Хищник")).thenReturn(List.of("Животные", "Птицы", "Рыба"));
```
А здесь animal каким образом участвует в тестируемом методе?

Shim239 replied:
Исправил, убрал мок

pomegranate282922 commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    @Test(expected = Exception.class)
+    public void doesHaveManeMethodExeptionBranch() throws Exception {
```
А разве именно doesHaveMane вызывает исключение? Тестовый метод назван некорректно и соответсвенно вторая строка метода лишняя.

Shim239 replied:
ай... две ветки в конструкторе про Mane, значит и третья про то же, изначально была такая логика :)
А вот то, что не нужен вызов метода lion.doesHaveMane() - это не заметил, да.. Из-за названия в т.ч.
Исправил, правда теперь как-то неестественно выглядит - создание неиспользуемого объекта класса Lion, ну что ж, такой тест.. )

pomegranate282922 reviewed (commented):
Хорошая работа! Есть ошибка при использовании подмены результата - поправь ее.
Далее в тестах на cat делаешь мок animal. Но сам этот объект где участвует - непонятно, живет рядом параллельной жизнью)
Ещё пара мелких замечаний - поправь их тоже, пожалуйста.

Shim239 committed:
исправления по результатам ревью;
- CatTest - исправлен мок
- FelineTest - убран мок
- LionTest - поправлен метод, нейминг по смыслу проверяемого

pomegranate282922 commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,50 @@
+        Lion lion = new Lion("Самка", mockFeline);
+        Mockito.when(mockFeline.getFood("Хищник")).thenReturn(List.of("Животные", "Птицы", "Рыба"));
+        List<String> actualResult = List.of("Животные", "Птицы", "Рыба");
+        List<String> expectedResult = lion.getFood();
```
Лучше дважды не создавать один и тот же список, а переиспользовать

pomegranate282922 reviewed (approved):
Отлично, зачёт!
